public/components/common/custom-search-bar/custom-search-bar.tsx

```diff
@@ -1,165 +1,190 @@
-import { FilterMeta, FilterState, FilterStateStore } from '../../../../../../src/plugins/data/common';
-
 import {
-    EuiFlexGroup,
-    EuiFlexItem,
-    EuiSwitch,
-} from '@elastic/eui';
-
-//@ts-ignore
+  FilterMeta,
+  FilterState,
+  FilterStateStore,
+} from '../../../../../../src/plugins/data/common';
+import { EuiFlexGroup, EuiFlexItem, EuiSwitch } from '@elastic/eui';
 import { getDataPlugin } from '../../../kibana-services';
 import { KbnSearchBar } from '../../kbn-search-bar';
-import { Combobox } from './components'
+import { Combobox, MultiSelect } from './components';
 
 export const CustomSearchBar = ({ filtersValues, ...props }) => {
-
-    const KibanaServices = getDataPlugin().query;
-    const filterManager = KibanaServices.filterManager;
-    const indexPattern = getIndexPattern();
-    const defaultSelectedOptions = () => {
-        const array = []
-        filtersValues.forEach(item => {
-            array[item.key] = []
-        })
-
-        return array
-    }
-    const [avancedFiltersState, setAvancedFiltersState] = useState(false);
-    const [selectedOptions, setSelectedOptions] = useState(defaultSelectedOptions);
-
-    useEffect(() => {
-        let filterSubscriber = filterManager.getUpdates$().subscribe(() => {
-            onFiltersUpdated()
-            return () => {
-                filterSubscriber.unsubscribe();
-            };
-        });
-    }, []);
-
-    const onFiltersUpdated = () => {
-        refreshCustomSelectedFilter()
-    }
-
-    const changeSwitch = () => {
-        setAvancedFiltersState(state => !state);
-    }
-
-
-    const buildCustomFilter = (isPinned: boolean, index?: any, querySearch?: any, key?: any): Filter => {
-        const meta: FilterMeta = {
-            disabled: false,
-            negate: false,
-            key: key,
-            params: { query: querySearch },
-            alias: null,
-            type: "phrase",
-            index,
-        };
-        const $state: FilterState = {
-            store: isPinned ? FilterStateStore.GLOBAL_STATE : FilterStateStore.APP_STATE,
-        };
-        const query = {
-            match_phrase: {
-                [key]: {
-                    query: querySearch
-                }
-            }
-        }
-
-        return { meta, $state, query };
+  const KibanaServices = getDataPlugin().query;
+  const filterManager = KibanaServices.filterManager;
+  const indexPattern = getIndexPattern();
+  const defaultSelectedOptions = () => {
+    const array = [];
+    filtersValues.forEach((item) => {
+      array[item.key] = [];
+    });
+
+    return array;
+  };
+  const [avancedFiltersState, setAvancedFiltersState] = useState(false);
+  const [selectedOptions, setSelectedOptions] = useState(defaultSelectedOptions);
+
+  useEffect(() => {
+    let filterSubscriber = filterManager.getUpdates$().subscribe(() => {
+      onFiltersUpdated();
+      return () => {
+        filterSubscriber.unsubscribe();
+      };
+    });
+  }, []);
```

Comment:
This `useEffect` does not properly cancel the subscription as the return statement is the subscriber's rather than `useEffect`'s
The hook `useFilterManager` already manages the subscription